docker.yaml

```diff
@@ -0,0 +1,141 @@
+  # TODO: Figure out a way to autobump this as required
+  # https://github.com/docker-library/docker/tree/master
+  - working-directory: /home/docker-library
+    pipeline:
+      - uses: fetch
+        with:
+          uri: https://github.com/docker-library/docker/archive/12a4554648b671111eaf4cf4636cb1d6310c65e1.zip
+          expected-sha256: 90386eae1de356036bb7b50129f0a214e299cc474ef3e305b41332fb2e2e6f81
+          extract: false
+      - runs: |
+          unzip -q 12a4554648b671111eaf4cf4636cb1d6310c65e1.zip
+          mv */** .
+
+  # Docker is vehemenetly against stripping the resulting binary
+  # Ref: https://github.com/moby/moby/blob/d8a51d2887cbc465ab8d76ed98f7a86996ab3c22/project/PACKAGERS.md#stripping-binaries
+  # - uses: strip
+  - runs: |
+      # this exists to appease yam
+
+subpackages:
+  - name: dockerd
+    description: "Docker Engine (dockerd)"
+    dependencies:
+      runtime:
+        # https://github.com/docker/docker/blob/master/project/PACKAGERS.md#runtime-dependencies
+        - btrfs-progs
+        - e2fsprogs
+        - e2fsprogs-extra
+        - ip6tables
+        - iptables
+        - openssl
+        - xfsprogs
+        - xz
+        - pigz
+        - zfs
+        - containerd
+        - ctr
+        - tini-static
+    pipeline:
+      - runs: |
+          install -Dm755 bundles/dynbinary-daemon/docker-proxy ${{targets.destdir}}/usr/bin/docker-proxy
+          install -Dm755 bundles/dynbinary-daemon/dockerd ${{targets.destdir}}/usr/bin/dockerd
+          ln -sf /sbin/tini-static ${{targets.destdir}}/usr/bin/docker-init
+
+  - name: docker-oci-entrypoint
+    description: "docker OCI entrypoint"
+    dependencies:
+      runtime:
+        - docker
+    pipeline:
+      - runs: |
+          install -Dm755 /home/docker-library/docker-entrypoint.sh "${{targets.subpkgdir}}"/usr/bin/docker-entrypoint.sh
+
+  - name: dockerd-oci-entrypoint
+    description: "dockerd OCI entrypoint"
+    dependencies:
+      runtime:
+        - docker
+        # Used as a fallback in the dockerd-entrypoint.sh script
+        - docker-oci-entrypoint
+        # Used as a fallback in the dockerd-entrypoint.sh script
+        - docker-dind
+    pipeline:
+      - runs: |
+          install -Dm755 /home/docker-library/dockerd-entrypoint.sh "${{targets.subpkgdir}}"/usr/bin/dockerd-entrypoint.sh
+
+  # Ref: https://github.com/docker-library/docker/blob/master/Dockerfile-dind.template
+  - name: docker-dind
+    description: "Docker in Docker"
+    dependencies:
+      runtime:
+        - docker
+    pipeline:
+      - runs: |
+          install -Dm755 /home/build/hack/dind "${{targets.subpkgdir}}"/usr/bin/dind
+
+  # Ref: https://github.com/docker-library/docker/blob/master/Dockerfile-git.template
+  - name: docker-git
+    description: "Docker in Docker with Git"
+    dependencies:
+      runtime:
+        - docker-dind
+        - git
```

Comment:
Does this need to be a meta-package? This seems like something we could just replicate at image-build-time.

Reply:
that's fair, I got carried away with the other `docker-*` and didn't stop to think. incoming